src/pytorch_lightning/strategies/ddp.py

```diff
@@ -453,9 +453,23 @@ def reconciliate_processes(self, trace: str) -> None:
         if len(os.listdir(sync_dir)) == (self.world_size // self.num_nodes):
             return
 
+        # Shutdown all other processes. First, we try sending a SIGINT since
+        # that triggers a KeyboardInterrupt exception by default, which should
+        # allow any context managers and try/finally logic to run.
         for pid in self._pids:
             if pid != os.getpid():
-                os.kill(pid, signal.SIGKILL)
+                os.kill(pid, signal.SIGINT)
```

Comment:
I see you tried to add an end to end test. But this can get really tricky. Think of this:

If your logic is correct and it sends the sigkill correctly, then that would mean that the test process testing this would be killed. But then this is indistinguishable from not running the test at all. 

To test it end-to-end, we would somehow need to run everything from a separate process that waits for the result, then checks whether it's subprocess, the actual test, has stopped and was killed. 

This seems too involved. So I'm suggesting a simpler test that mocks the `os.kill()` call here. We can then verify that the mock was called correctly. Both worker processes would exit the simulation with success. What do you think?